python_ta/z3/z3_parser.py

```diff
@@ -15,37 +15,26 @@ class Z3ParseException(Exception):
     pass
 
 
-class ExprWrapper:
+class Z3Parser:
     """
-    Wrapper class to convert an astroid expression node into a z3 expression.
+    Convert an astroid expression node into a z3 expression.
 
     Instance attributes:
-        - node: astroid node obtained given by the value attribute of astroid expression.
         - types: dictionary mapping variable names in astroid expression to their type name or z3 variable.
     """
 
-    node: astroid.NodeNG
     types: Dict[str, Union[str, z3.ExprRef]]
 
-    def __init__(self, node: astroid.NodeNG, types=None):
+    def __init__(self, types=None):
         if types is None:
             types = {}
         self.types = types
 
-        # extract expression out of expression statement
-        if isinstance(node, (astroid.Expr, astroid.Assign)):
-            self.node = node.value
-        else:
-            self.node = node
-
-    def reduce(self, node: astroid.NodeNG = None) -> z3.ExprRef:
+    def parse(self, node: astroid.NodeNG = None) -> z3.ExprRef:
```

Comment:
Now that there's no `self.node` instance attribute, it doesn't make sense to have `node` be an optional argument---it should always be required.